src/D2L.Bmx/ConsolePrompter.cs

```diff
@@ -112,27 +112,29 @@ internal class ConsolePrompter : IConsolePrompter {
 		return roles[index - 1];
 	}
 
-	int IConsolePrompter.PromptMfa( MfaOption[] mfaOptions ) {
+	OktaMfaFactor IConsolePrompter.SelectMfa( OktaMfaFactor[]? mfaOptions ) {
 		Console.Error.WriteLine( "MFA Required" );
-		if( mfaOptions.Length > 1 ) {
-			for( int i = 0; i < mfaOptions.Length; i++ ) {
-				Console.Error.WriteLine( $"[{i + 1}] {mfaOptions[i].Provider}: {mfaOptions[i].Name}" );
-			}
-			Console.Error.Write( "Select an available MFA option: " );
-			if( !int.TryParse( Console.ReadLine(), out int index ) || index > mfaOptions.Length || index < 1 ) {
-				throw new BmxException( "Invalid account selection" );
-			}
-			return index;
-		} else if( mfaOptions.Length == 0 ) {//idk, is mfaOptions' length guaranteed to be >= 1?
+
+		if( mfaOptions is not { Length: > 0 } ) {
```

Comment:
this catches both null and zero length `mfaOptions`
(but actually soon we won't need the null check because I'll update code in OktaApi to never return a nullable MFA factor array)

Reply:
TIl, cool syntax. It does border a little on being too clever. Seems fine for this use case though.

Reply:
From what I've seen on the Internet, pattern matching in C# is freaking divisive...
Many people just say it's "not readable", which I interpret to be "I haven't learned the new pattern matching syntax".
I'm obviously in the pro pattern matching group though 😛

Reply:
Yeah I am guilty of that, re: rejecting concise syntax because its unfamiliar rather than because its actually too cheeky. 😆